source/jormungandr/jormungandr/schedule.py

```diff
@@ -256,11 +261,21 @@ def next_departures(self, request):
                             for rp in resp.route_points)
 
         rt_proxy = None
+
+        futures = []
+        pool = gevent.pool.Pool(app.config.get('GREENLET_RT_POOL_SIZE', 3))
+
+        def worker(rt_proxy, route_point, template, request, resp):
+            return resp, rt_proxy, route_point, template, self._get_next_realtime_passages(rt_proxy, route_point, request)
+
         for route_point, template in route_points.items():
             rt_proxy = self._get_realtime_proxy(route_point)
             if rt_proxy:
-                next_rt_passages = self._get_next_realtime_passages(rt_proxy, route_point, request)
-                rt_proxy._update_passages(resp.next_departures, route_point, template, next_rt_passages)
+                futures.append(pool.spawn(worker, rt_proxy, route_point, template, request, resp))
```

Comment:
You will probably have to pass the `request_id` too

Reply:
As discussed, the request id can't be used in greenlet as is

Reply:
The point was to find a way to have it in the greenlet, I won't merge it without it. It's far too valuable for debuging purpose.